Create FST-1029-langversion and preview library functions.md (#422)
fsharp/fslang-design: tooling/FST-1029-langversion and preview library functions.md

The design suggestion [](https://github.com/dotnet/fsharp/issues/5496) has been marked "approved in principle".
This RFC covers the detailed proposal for this suggestion.

* [ ] Approved in principle
* [ ] [Suggestion](https://github.com/fsharp/fslang-suggestions/issues/825)

> * [ ] Implementation: [In progress]https://github.com/dotnet/fsharp/pull/8042)

Reviewer: ```suggestion
* [ ] Implementation: [In progress](https://github.com/dotnet/fsharp/pull/8042)
```

# Summary

> [summary]: Allow --langversion:preview to disable compiler warnings from APIs using the ExperimentalAttribute().

Reviewer: ```suggestion
Allow --langversion:preview to disable compiler warnings from APIs using the ExperimentalAttribute().
```

# Motivation

> [motivation]: Shipping preview versions of the F# language early is very difficult, because there isn't a delivery mechanism that is seperate from Visual Studio or the DotNet Cli.

Reviewer: ```suggestion
Shipping preview versions of the F# language early is very difficult, because there isn't a delivery mechanism that is separate from Visual Studio or the .NET CLI.
```

> C# has the same issue and has chosen to ship preview C# lang features in the RTM builds, concealed by the --langversion switch.  F# also has a --langversion switch, however, many F# compiler features rely on new APIs added to FSharp.Core.

Reviewer: ```suggestion
C# has the same issue and has chosen to ship preview C# lang features in the RTM builds, concealed by the --langversion switch.  F# also has a --langversion switch, however, many F# compiler features rely on new APIs added to FSharp.Core.
```

> e.g. nameof and the 3D and 4D slicing enhancments APIs.

Reviewer: ```suggestion
For example,  `nameof` and the 3D and 4D slicing enhancement APIs all reside in FSharp.Core with no good way to let developers know that these are a part of a preview language feature.
```

> This feature will allow us to ship preview library features in the FSharp.Core dll, in such a way that developers will have a warning if they try to use them from a compiler without the preview switch specified, but will not have to worry about warnings when preview is specified.

Reviewer: ```suggestion
This feature will allow us to ship preview library features in the FSharp.Core, in such a way that developers will have a warning if they try to use them from a compiler without the preview switch specified, but will not have to worry about warnings when preview is specified.
```

# Detailed design

> [design]: 

Reviewer: ```suggestion
```
